includes/class-wc-order.php

```diff
@@ -103,25 +108,32 @@ public function payment_complete( $transaction_id = '' ) {
 			if ( ! $this->get_id() ) {
 				return false;
 			}
-			do_action( 'woocommerce_pre_payment_complete', $this->get_id() );
 
-			if ( ! empty( WC()->session ) ) {
-				WC()->session->set( 'order_awaiting_payment', false );
-			}
+			$lock_name = 'order_' . $this->get_id() . '_payment_complete';
+
+			if ( WC_Database_Locks::aquire_lock( $lock_name ) ) {
```

Comment:
@claudiulodro No, it returns false. Only true once a lock is created successfully: https://github.com/woocommerce/woocommerce/pull/15967/files#diff-e63c6a8e54cbd89064a6f330abf1248cR107